airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -21,67 +21,201 @@
+        """
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+        zero_date = datetime(1900, 1, 1)
+
+        current_state = current_stream_state.get(self.cursor_field)
+        latest_state = latest_record.get(self.cursor_field)
+        if not any([current_state, latest_state]):  # noqa
+            return {self.cursor_field: zero_date}
+
+        if not all([current_state, latest_state]):
+            nonzero_state = list(filter(bool, [current_state, latest_state]))[0]
+            if not isinstance(nonzero_state, datetime):
+                nonzero_state = dateutil.parser.isoparse(nonzero_state)
+            return {self.cursor_field: nonzero_state}
+        else:
+            if not isinstance(current_state, datetime):
+                current_state = dateutil.parser.isoparse(current_state)
+            if not isinstance(latest_state, datetime):
+                latest_state = dateutil.parser.isoparse(latest_state)
+            max_value = max(latest_state, current_state)
+            return {self.cursor_field: max_value}
+
+    def request_params(self, stream_state=None, **kwargs):
+
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        default_params = {"sort_field": self.cursor_field, "sort_dir": "ASC"}
+        since_value = stream_state.get(self.cursor_field)
+        if isinstance(since_value, datetime):
+            since_value = since_value.isoformat()
+        default_params[f"since_{self.cursor_field}"] = since_value
+        params.update(default_params)
+        return params
+
+
+class Lists(IncrementalMailChimpStream):
+    cursor_field = "date_created"
+    name = "Lists"
+    data_field = "lists"
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class Campaigns(IncrementalMailChimpStream):
+    cursor_field = "create_time"
+    name = "Campaigns"
+    data_field = "campaigns"
+
+    def path(self, **kwargs) -> str:
+        return "campaigns"
+
+
+class EmailActivity(MailChimpStream):
+    cursor_field = "timestamp"
+    name = "Email_activity"
+    data_field = "emails"
+
+    def stream_slices(self, **kwargs):
+        campaign_stream = Campaigns(authenticator=self.authenticator)
+        for campaign in campaign_stream.read_records(sync_mode=SyncMode.full_refresh):
+            yield {"campaign_id": campaign["id"]}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        campaign_id = stream_slice["campaign_id"]
+        return f"reports/{campaign_id}/email-activity"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
```

Comment:
not sure, why we are touching stream_state there, are you sure this stream is not incremental? (not inherited from `IncrementalStream`)
if it is so then keep in mind that slicing of stream should be reflected in its state as well, because for each slice (campaign in this case) you want to have separate state (timestamp) saved, so the state structure will be:
```json
{
  "email_activities": {
    "<campaign_id>": {
    "since": "<your_datetime>"
  }
}
```